flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/util/WindowEmitStrategy.scala

```diff
@@ -128,12 +151,59 @@ object WindowEmitStrategy {
       // use min idle state retention time as allow lateness
       tableConfig.getMinIdleStateRetentionTime
     }
-
+    val enableEarlyFireDelay = tableConfig.getParameters.getBoolean(
+      SQL_EXEC_EMIT_EARLY_FIRE_ENABLED)
+    val earlyFireDelay = getMillisecondFromConfigDuration(
+      tableConfig, SQL_EXEC_EMIT_EARLY_FIRE_DELAY)
+    val enableLateFireDelay = tableConfig.getParameters.getBoolean(SQL_EXEC_EMIT_LATE_FIRE_ENABLED)
+    val lateFireDelay = getMillisecondFromConfigDuration(
+      tableConfig, SQL_EXEC_EMIT_LATE_FIRE_DELAY)
     new WindowEmitStrategy(
       isEventTime,
       isSessionWindow,
-      tableConfig.getEarlyFireInterval,
-      tableConfig.getLateFireInterval,
+      earlyFireDelay,
+      enableEarlyFireDelay,
+      lateFireDelay,
+      enableLateFireDelay,
       allowLateness)
   }
+
+  // It is a experimental config, will may be removed later.
+  @Experimental
+  val SQL_EXEC_EMIT_EARLY_FIRE_ENABLED: ConfigOption[Boolean] =
```

Comment:
I think the plan was to delete these configs, making we only have watermark trigger strategy

Reply:
We have supported early-fire and late-fire, and have tests to cover that. I think we need some way to configure the emit strategy if we want to keep the tests.

Reply:
@KurtYoung @wuchong yes, I keep those configs because there are tests to cover early emit and late emit.

Reply:
ok, let's keep it internal then.